### DISCUSSION TITLE
support for bigint?

### DISCUSSION THREAD

**Discussion by robifr** [d3/d3#3849] [Q&A] [answered]
Is there a plan to support bigint? I'm currently use this library on android project, with a requirement to display a large axis and bar chart

```js
d3.scaleLinear()
      .domain([5_223_372_036_854_775_807n, 9_223_372_036_854_775_807n])
      .range([BigInt(minRange), BigInt(maxRange)]) //Uncaught TypeError: Cannot convert a BigInt value to a number
```

**Comment by Fil** [comment, marked as answer]
That's not a goal, and JavaScript doesn't make it an easy fix. I'm afraid you'll have to handle conversion before you call d3.scaleLinear.

**Reply by robifr** [reply]
yeah, my current solution is by converting the big number into percentage values and replace every numbers in percent with an array of strings. bit inconvenience, but works perfectly.
